sleap_nn/architectures/convnext.py

```diff
@@ -0,0 +1,216 @@
+"""This module provides a generalized implementation of ConvNext.
+
+See the `ConvNextWrapper` class docstring for more information.
+"""
+
+from functools import partial
+from typing import Any, Callable, List, Optional, Dict, Tuple
+
+import torch
+from torch import nn, Tensor
+from torchvision.ops.misc import Conv2dNormActivation
+from torchvision.utils import _log_api_usage_once
+from sleap_nn.architectures.encoder_decoder import Decoder
+from torchvision.models.convnext import LayerNorm2d, CNBlock, CNBlockConfig
+
+
+class ConvNeXtEncoder(nn.Module):
+    """Src: torchvision.models.
```

Comment:
This is not an appropriate class docstring first line. It should be a 1-line summary of the class.